### PR TITLE
TW-997 Return swap page on IOS

### PR DESCRIPTION
**Pull request by lendihop**
https://madfish.atlassian.net/jira/software/projects/TW/boards/4?selectedIssue=TW-997

### CODE REVIEW

File: src/navigator/navigation-bar/side-bar/side-bar-button/side-bar-button.tsx [hunk 1]
```diff
@@ -52,7 +51,12 @@ export const SideBarButton: FC<Props> = ({
     if (disabled) {
       disabledOnPress();
     } else {
-      navigate(routeName);
+      if (onSwapButtonPress) {
+        onSwapButtonPress();
+      } else {
+        // @ts-ignore
+        navigate(routeName, params);
```
Comment (keshan3262) [inline src/navigator/navigation-bar/side-bar/side-bar-button/side-bar-button.tsx:58]: Sorry, using `@ts-ignore` looks a bit shitty here. Please try finding a type or writing a new one like
```
interface NavigationArguments<T extends keyof NavigationParamList> {
  routeName: T;
  params: NavigationParamList[T];
}
```
then passing such value to this component and using it here. As an alternative, you may try using `NavigationArgsType`.

File: src/navigator/navigation-bar/tab-bar/tab-bar-button/tab-bar-button.tsx [hunk 2]
```diff
@@ -50,8 +52,12 @@ export const TabBarButton: FC<Props> = ({
     if (disabled) {
       disabledOnPress();
     } else {
-      // @ts-ignore
-      navigate(routeName, params);
+      if (onSwapButtonPress) {
+        onSwapButtonPress();
+      } else {
+        // @ts-ignore
+        navigate(routeName, params);
```
Comment (keshan3262) [inline src/navigator/navigation-bar/tab-bar/tab-bar-button/tab-bar-button.tsx:59]: If you succeed, use the same approach here

**Review by herkoss — APPROVED** [review]
GJ!

**Review by keshan3262 — APPROVED** [review]

**Review by herkoss — APPROVED** [review]

**Review by IrynaKhyzhynska — APPROVED** [review]
android done 1.15_879

**Review by vladvn1905 — DISMISSED** [review]
ios done build: 1.15_879

**Pull request merged**